xds/internal/clients/xdsclient/xdsconfig.go

```diff
@@ -0,0 +1,84 @@
+/*
+ *
+ * Copyright 2025 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package xdsclient
+
+import (
+	"google.golang.org/grpc/xds/internal/clients"
+)
+
+// Config is used to configure an xDS client. After one has been passed to the
+// xDS client's New function, it must not be modified. A Config may be reused;
+// the xdsclient package will also not modify it.
+type Config struct {
+	// Servers specifies a list of xDS management servers to connect to. The
+	// order of the servers in this list reflects the order of preference of
+	// the data returned by those servers. xDS client use the first
+	// available server from the list.
+	//
+	// See gRFC A71 for more details on fallback behavior when the primary
+	// xDS server is unavailable.
+	//
+	// gRFC A71: https://github.com/grpc/proposal/blob/master/A71-xds-fallback.md
+	Servers []clients.ServerConfig
+
+	// Authorities map is used to define different authorities, in a federated
+	// setup, each with its own set of xDS management servers.
+	Authorities map[string]Authority
+
+	// Node is the identity of the xDS client connecting to the xDS
+	// management server.
+	Node clients.Node
+
+	// TransportBuilder is used to connect to the xDS management server.
+	TransportBuilder clients.TransportBuilder
+
+	// ResourceTypes is a map from resource type URLs to resource type
+	// implementations. Each resource type URL uniquely identifies a specific
+	// kind of xDS resource, and the corresponding resource type implementation
+	// provides logic for parsing, validating, and processing resources of that
+	// type.
+	ResourceTypes map[string]ResourceType
+}
+
+// Authority contains configuration for an xDS control plane authority.
+//
+// See: https://github.com/grpc/grpc/blob/master/doc/grpc_xds_bootstrap_format.md
```

Comment:
This link doesn't seem that helpful to me -- at least, it's a big-ish doc with a bunch of other stuff in it, and doesn't link to a section about the topic.

It doesn't seem like anyone actually uses federation, because I can't find any good documentation on it, either.

This is about all I can find in Envoy's docs, too:

https://github.com/cncf/xds/blob/main/proposals/TP1-xds-transport-next.md

https://www.envoyproxy.io/docs/envoy/latest/xds/core/v3/resource_locator.proto#xds-core-v3-resourcelocator